dashboard/app/controllers/application_controller.rb

```diff
@@ -154,6 +154,12 @@ def milestone_response(options)
       end
     end
 
+    if PuzzleRating.enabled?
+      if script_level && PuzzleRating.can_rate?(script_level.script, script_level.level, current_user)
```

Comment:
It's basically an exists check with a few other conditions. Is CanCan flexible enough for that?